src/Native/Runtime/amd64/WriteBarriers.asm

```diff
@@ -425,6 +425,16 @@ LEAF_ENTRY RhpByRefAssignRef, _TEXT
     cmp     rcx, [g_ephemeral_high]
     jae     RhpByRefAssignRef_NotInHeap
 
+
+ifdef CORERT
+    ;; On CoreRT, JIT uses this helper to update stack locations.
+    ;; Check whether the writes were even into the heap. If not there's no card update required.
+    cmp     rdi, [G_LOWEST_ADDRESS]
+    jb      RhpByRefAssignRef_NotInHeap
+    cmp     rdi, [G_HIGHEST_ADDRESS]
```

Comment:
Nit - there is mix of ALL_CAPS and lowercase references to GC globals in this file. It may be nice to use lowercase everywhere in this file, and in ARM version as well.